### ISSUE TITLE
CPU usage alarm getting triggered below the threshold

### ISSUE THREAD

**Issue by rrreche** [firehol/netdata#2165]
Hello,

I have just started using this software. Amazing project, loving it so far. Thank you.

There is one thing though that I cannot get through. I have configured a custom alarm that monitors CPU - user, system and softirq. For the most part, I have copied it from the cpu.conf example. I have removed all the health.d directory and left only this cpu.conf with the following content:

```
template: media_consumo_ultimo_minuto
      on: system.cpu
  lookup: average -1m unaligned of user,system,softirq
   units: %
   every: 1m
   warn: $this > (($status >= $WARNING)  ? ($green) : ($red))
   crit: $this > (($status == $CRITICAL) ? ($red) : ($red + 10))
   delay: up 30s down 30s multiplier 1.0 max 1m
    info: Consumo medio del ultimo minuto por encima de los umbrales normales
      to: sysadmin
   green: 60
     red: 70
   exec: /usr/libexec/netdata/plugins.d/alarm-cpu.sh
```

As I understand it, this alarm should not be triggered unless my average of combined cpu.user+system+softirq is above the green threshold, which is 60. But I am receiving WARN and CRIT notifications on 40% and 50% of usage, respectively. Any hint on what could I do to troubleshoot this?

**Comment by ktsaou** [comment]
Hi, thanks!

On the dashboard, there is a badge for this alarm (on the alarm dialog). It shows `$this`. What value does it show?

The alarm will calculate the 1 minute average, of the **sum** of user, system and softirq for each data collection. So, for each of the last 60 seconds netdata will find the sum of the 3 dimensions - this will result in a new dimension with 60 entries, and then it will calculate the average  of the combined dimension for these 60 seconds.

To check what netdata does, you can do these:

1. edit netdata.conf and set `debug flags = 0x0000000000800000` ([what is this value?](https://github.com/firehol/netdata/blob/855aa2680dd26670af125978179d1d3db9211557/src/log.h#L27)

2. compile and install netdata like this: `CFLAGS="-O2 -DNETDATA_INTERNAL_CHECKS=1" ./netdata-installer.sh`

3. now in `/var/log/netdata/debug.log` will see a lot of output about everything netdata does with each alarms

Of course, undo the changes after you are done.

**Comment by rrreche** [comment]
Thank you for your early reply! I think I have this almost nailed. I have another alarm in my cpu.conf:

> template: media_iowait_ultimos_5_minutos
>       on: system.cpu
>   lookup: average -30s unaligned of iowait
>    units: %
>    every: 1m
>     warn: $this > $green && $this < $red
>     crit: $this > $red
>    delay: up 30s down 30s multiplier 1.0 max 1m
>     info: Aumento de las operaciones de lectura/escritura en disco. Posible cuello de botella
>       to: sysadmin
>    green: 20
>      red: 40
>    exec: /usr/libexec/netdata/plugins.d/alarm-iowait.sh

As you pointed out, I went to check the badges, and also the alarm GUI in the webserver. Interesting enough, it seems like the green and red thresholds on this iowait alarm are replacing the ones configured in the cpu usage. I tried putting them in different files, but still, they seem to share their thresholds

![imagen](https://cloud.githubusercontent.com/assets/26229304/25809363/fb711e14-340c-11e7-82ef-4b8a0c4157c6.png)

Removing the iowait alarm restores the threshold for the cpu usage. Is this the intended behavior? Are the thresholds shared? And if so, which the sharing criteria, maybe per family (system.cpu)?
